sihl_contrib/session/lib/repo/postgresql/repo.ml

```diff
@@ -1 +1,98 @@
-(* TODO *)
+module Sql = struct
+  module Session = struct
+    open Sihl_session.Model.Session
+
+    let get_all =
+      [%rapper
+        get_many
+          {sql|
+        SELECT
+          session_key as @string{key},
+          session_data as @string{data},
+          @pdate{expire_date}
+        FROM sessions_sessions
+        |sql}
+          record_out]
+
+    let get =
+      [%rapper
+        get_opt
+          {sql|
+        SELECT
+          session_key as @string{key},
+          session_data as @string{data},
+          @pdate{expire_date}
+        FROM sessions_sessions
+        WHERE session_sessions.key = %string{key}
+        |sql}
+          record_out]
+
+    let upsert =
+      [%rapper
+        execute
+          {sql|
+        INSERT INTO sessions_sessions (
+          session_key,
+          session_data,
+          expire_date
+        ) VALUES (
+          %string{key},
+          %string{data},
+          %pdate{expire_date}
+        ) ON CONFLICT (key) DO UPDATE SET
+        session_data = %string{data}
+        |sql}
+          record_in]
```

Comment:
Having `upsert` in the repo API will make it very hard to go from this to change set approach in the future. On the other hand, we could have the change set API in the service level, but I'm not sure if that will cause some problematic inefficiencies. We should talk about this a bit.

Reply:
Actually today I had a 3h bug because of upsert. It is hard to tell when an insert fails if it can fall back to upsert, so let's just split these two out of security reasons.
I think changesets should be on either service level or the database should be based on that kind of thinking (Datomic).